File: tree/bst.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std ;
struct Node{
    int data;
    struct Node* left;
    struct Node* right;
    Node(int x){
        data=x;
        left=right=NULL;
    }
};
struct Node* insert(struct Node* root,int key){
if(root==NULL){
    struct Node* l=new Node(key);
    return l;
}
else if(key<root->data){
root->left=insert(root->left,key);
}
else{
    root->right=insert(root->right,key);
}
return root;

}
void inorder(struct Node* j){
    if(j!=NULL){
        inorder(j->left);
        cout<<j->data<<" ";
        inorder(j->right);
    }
}
int main(){
int n;
cin>>n;
struct Node* root=NULL;
root=insert(root,10);
// for(int i=0;i<n-1;i++){
//     int n1;
//     cin>>n1;
//     insert(root,n1);
// }
insert(root,5);
insert(root,4);
insert(root,6);
insert(root,15);
insert(root,12);
insert(root,16);
inorder(root);

return 0;
}

File: bst.java
import java.util.*;
class Node{
    int val;
    Node left;
    Node right;
    Node(int c){
        val=c;
        left=right=NULL;
    }
}
class BST{
    Node root;
    BST(){
        root=NULL;
    }
    public static Node createTree(){
        int n;
        Scanner sc-new Scanner(System.in);
        n=sc.nextInt();
        for(int i=0;i<n;i++){
            root=insert(root,sc.nextInt());
        }
    }
    public static Node insert(Node root,int val)
    {

        if(root== NULL){
            root=new Node(val);
        }if(root.val<val){
            root.right=insert(root,val);
        }
        else if(root.val>val){
            root.left=insert(root,val);
        }
        return root;
    }
    public static void inorder(Node root){
        if(root==NULL)return ;
        inorder(root->left);
        System.out.print(root.val+" ");
        inorder(root.right);
    }

};
class main(string args[]){
     
}